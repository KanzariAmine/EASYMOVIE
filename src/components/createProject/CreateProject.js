import React, { useState, useContext } from "react";
import {
  Header,
  Container,
  Form,
  FormControl,
  Textarea,
  Button,
} from "./CreateProject.styled";
import { ProjectContext } from "../../context/ProjectContext";

function CreateProject() {
  const { dispatch } = useContext(ProjectContext);

  const [project, setProject] = useState({
    nom: "",
    description: "",
  });

  const handelChange = (e) => {
    e.preventDefault();
    setProject({ ...project, [e.target.name]: e.target.value });
  };

  const handelSubmit = (e) => {
    e.preventDefault();
    dispatch({ type: "ADD_PROJECT", project });
    setProject({ nom: "", description: "" });
  };

  return (
    <Container>
      <Header>
        <h2>Ajouter Project</h2>
      </Header>
      <Form onSubmit={handelSubmit}>
        <FormControl>
          <label>Nom</label>
          <input
            type="text"
            placeholder="Ajouter le nom du project"
            id="nom"
            name="nom"
            value={project.nom}
            onChange={handelChange}
          />
        </FormControl>
        <FormControl>
          <label>Description</label>
          <Textarea
            type="text"
            placeholder="Descriptiont..."
            name="description"
            value={project.description}
            onChange={handelChange}
          />
        </FormControl>
        <Button>Ajouter</Button>
      </Form>
    </Container>
  );
}

export default CreateProject;
