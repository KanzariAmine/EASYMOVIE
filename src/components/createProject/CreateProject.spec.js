import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import CreateProject from "./CreateProject";
import ProjectContextProvider, {
  ProjectContext,
} from "../../context/ProjectContext";

const renderWithContext = (component) => {
  return {
    ...render(
      <ProjectContextProvider value={ProjectContext}>
        {component}
      </ProjectContextProvider>
    ),
  };
};

afterEach(cleanup);

describe("Test Create Project Component", () => {
  test("Should take snapshot", () => {
    const { asFragment } = renderWithContext(<CreateProject />);
    expect(asFragment(<CreateProject />)).toMatchSnapshot();
  });
});
