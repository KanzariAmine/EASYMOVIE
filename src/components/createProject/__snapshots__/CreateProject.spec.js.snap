// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Create Project Component Should take snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hTciaQ"
  >
    <div
      class="sc-AxirZ lnfslJ"
    >
      <h2>
        Ajouter Project
      </h2>
    </div>
    <form
      class="sc-AxiKw eyRPdh"
    >
      <div
        class="sc-AxhCb hgRKkH"
      >
        <label>
          Nom
        </label>
        <input
          id="nom"
          name="nom"
          placeholder="Ajouter le nom du project"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="sc-AxhCb hgRKkH"
      >
        <label>
          Durée
        </label>
        <input
          name="duree"
          placeholder="Durée du vidéo minutes..."
          required=""
          type="number"
          value=""
        />
      </div>
      <div
        class="sc-AxhCb hgRKkH"
      >
        <label>
          Description
        </label>
        <textarea
          class="sc-AxhUy cQVZIA"
          name="description"
          placeholder="Descriptiont..."
          required=""
          type="text"
        />
      </div>
      <button
        class="sc-AxgMl dYYlGh"
      >
        Ajouter
      </button>
    </form>
  </div>
</DocumentFragment>
`;
